Automattic/wp-calypso
# Marketplace: Add Reviews Rating component to the reviews form

Related to #https://github.com/Automattic/dotcom-forge/issues/4661

## Proposed Changes

* Add ReviewsRatingsStars component to Calypso components. This component has been copied and updated from [WooCommerce component](https://github.com/Automattic/woocommerce.com/blob/a1f6b43a44b4f8c8a244586ad1ee371c82aaee8a/themes/woo/components/reviews/ratings-stars/ratings-stars.js)
* Use that component in the ReviewsModal to introduce the ratings
* Create a ReviewsSummary component that contains the reviews area in the Theme details screen


|Before | After|
|-------|------|
|      ![CleanShot 2023-12-05 at 19 46 56@2x](https://github.com/Automattic/wp-calypso/assets/3519124/3fb28000-3d94-4d73-9af0-5c86b968ce2f)   |       ![CleanShot 2023-12-05 at 19 46 16@2x](https://github.com/Automattic/wp-calypso/assets/3519124/766d4b92-b72e-45d8-81e6-a1a04dce51b5)            |

## Testing Instructions



* With the flag `marketplace-add-review` enabled
* Navigate to a theme details page, e.g. `/themes/ hevor/:siteId` selecting any site
* Click on the Add Review button
* Make sure you see the ReviewsRatingsStars component 
* Select a rating choosing the number of stars
* Make sure the component allows to select different ratings and updates accordingly
* Click on Submit
* Make sure the network call `https://public-api.wordpress.com/wp/v2/sites/marketplace.wordpress.com/comments` contains the selected rating and it completes as expected. Please note that if there is a validation error in the backend, an error will be returned.



## Pre-merge Checklist
